packages/core/src/tabs/tab-group.ts

```diff
@@ -0,0 +1,248 @@
+import type { CSSResultGroup, TemplateResult } from 'lit';
+import type { EventEmitter } from '../utils/decorators/event.js';
+import type { ScopedElementsMap } from '@open-wc/scoped-elements';
+import { ScopedElementsMixin } from '@open-wc/scoped-elements';
+import { LitElement, html } from 'lit';
+import { property, queryAssignedElements, state } from 'lit/decorators.js';
+import { event } from '../utils/decorators/event.js';
+import { RovingTabindexController } from '../utils/controllers/roving-tabindex.js';
+import { Tab } from './tab.js';
+import styles from './tab-group.scss.js';
+import { TabPanel } from './tab-panel.js';
+
+let tabGroupCount = 0;
+
+export class TabGroup extends ScopedElementsMixin(LitElement) {
+  /** @private */
+  static get scopedElements(): ScopedElementsMap {
+    return {
+      'sl-tab': Tab,
+      'sl-tab-panel': TabPanel
+    };
+  }
+
+  /** @private */
+  static override styles: CSSResultGroup = styles;
+
+  /**
+   * Unique ID for each tab group component present.
+   */
+  private tabGroupId = `sl-tab-group-${tabGroupCount++}`;
+
+  private observer?: MutationObserver;
+
+  private static observerOptions = {
+    attributes: true,
+    subtree: true,
+    attributeFilter: ['selected'],
+    attributeOldValue: true
+  };
+
+  /** The slotted tabs. */
+  @queryAssignedElements({ slot: 'tabs' }) tabs?: Tab[];
+
+  #rovingTabindexController = new RovingTabindexController<Tab>(this, {
+    focusInIndex: (elements: Tab[]) => elements.findIndex(el => el.selected),
+    elements: () => this.tabs || [],
+    isFocusableElement: (el: Tab) => !el.disabled
+  });
+
+  /**
+   * The current tab node selected in the tab group.
+   */
+  @state() private selectedTab: Tab | null = this.initialSelectedTab;
+
+  @event() tabChange!: EventEmitter<number>;
+
+  @property({ reflect: true }) vertical = false;
+
+  /**
+   * Get the selected tab button, or the first tab button.
+   */
+  private get initialSelectedTab(): Tab | null {
+    return this.querySelector('sl-tab[selected]') || this.querySelector('sl-tab:not([disabled])');
+  }
+
+  override render(): TemplateResult {
+    return html`<div @click=${this.handleTabChange} role="tablist" @keydown=${this.handleKeydown} part="tab-list">
+        <span class="indicator" role="presentation"></span>
+        <slot name="tabs" @slotchange=${() => this.#rovingTabindexController.clearElementCache()}></slot>
+      </div>
+      <slot></slot>`;
+  }
+
+  override connectedCallback(): void {
+    super.connectedCallback();
+    this.updateSlots();
+  }
+
+  override firstUpdated(): void {
+    this.observer = new MutationObserver(this.handleMutation);
+    this.observer?.observe(this, TabGroup.observerOptions);
+    setTimeout(() => this.#updateSelectionIndicator(), 100);
+  }
+
+  private updateSlots(): void {
```

Comment:
Please use `#updateSlots()` instead of `private` TS keyword here as well (and below).

Reply:
I guess this goes for all private functions and variables?